src/components/AudioTrack/AudioTrack.vue

```diff
@@ -1,106 +1,51 @@
 <template>
   <div
     class="audio-track"
-    aria-label="$t('audio-track.aria-label')"
+    :aria-label="$t('audio-track.aria-label')"
     role="region"
   >
-    <!-- Only visible in compact player -->
-    <div v-if="isCompact" class="info-section flex justify-between">
-      <i18n path="audio-track.title" tag="p">
-        <template #title>
-          <NuxtLink to="#" class="text-pink hover:text-pink hover:underline">
-            {{ audio.title }}</NuxtLink
-          >
-        </template>
-        <template #creator>{{ audio.creator }}</template>
-      </i18n>
-      {{ audio.category }}
-    </div>
-
-    <div class="interactive-section flex flex-row gap-2">
-      <PlayPause
-        v-if="isCompact"
-        class="flex-shrink-0"
-        :is-playing="isPlaying"
-        :disabled="!isReady"
-        @toggle="setPlayerState"
-      />
-      <div
-        class="flex-grow"
-        @keypress.enter="setPlayerState(!isPlaying)"
-        @keypress.space="setPlayerState(!isPlaying)"
-      >
-        <Waveform
-          :class="isCompact ? 'h-20' : 'h-30'"
-          :message="message ? $t(`audio-track.messages.${message}`) : null"
-          :current-time="currentTime"
-          :duration="duration"
-          :peaks="audio.peaks"
-          :show-duration="isCompact"
-          @seeked="setPosition"
+    <Component
+      :is="layoutComponent"
+      :audio="audio"
+      :size="size"
+      :aspect="aspect"
+    >
```

Comment:
This is very cool, like component polymoprhism 😁

Reply:
Thanks 😃!